databroker/queries.py

```diff
@@ -98,6 +98,47 @@ def ScanID(*scan_ids, duplicates="latest"):
     return _ScanID(scan_ids=scan_ids, duplicates=duplicates)
 
 
+@register(name="scan_id_range")
+@dataclass
+class _ScanIDRange:
+    """
+    Find scans in the range.
+    """
+
+    start_id: int
+    end_id: int
+    duplicates: Duplicates
+
+    def __init__(self, *, start_id, end_id, duplicates):
+        self.start_id = start_id
+        self.end_id = end_id
+        self.duplicates = Duplicates(duplicates)
+
+    def encode(self):
+        return {
+            "start_id": self.start_id,
+            "end_id": self.end_id,
+            "duplicates": self.duplicates.value,
+        }
+
+    @classmethod
+    def decode(cls, *, start_id, end_id, duplicates):
+        return cls(
+            start_id=int(start_id),
+            end_id=int(end_id),
+            duplicates=Duplicates(duplicates),
+        )
+
+
+def ScanIDRange(start_id, end_id, duplicates="latest"):
+    # Wrap _ScanIDRange to provide a nice usage for *one or more scan_ids*:
```

Comment:
`ScanID` accepts variadic arguments, i.e. `ScanID(*scan_ids)`, so that we can one `ScanID(5)` or several `ScanID(5, 13, 21)` without doing this: `ScanID([5])`.

Variadic arguments don't play well with schemas, so I made

```py
_ScanID(scan_ids)  # must be a list
```

wrapped by

```py
ScanID(*scan_ids)  # variadic
```

That pattern is not necessary for `ScanIDRange` because there is no use case for variadic arguments here. You can just make one class, `ScanIDRange`, without the wrapping.